tests/unit/controller/http-client.test.ts

```diff
@@ -0,0 +1,130 @@
+        return HttpResponse.json({
+            data: "delete-test",
+        });
+    }),
+
+    http.get(`${PUBLIC_API_BASE_URL}`, () => {
+        return HttpResponse.json({
+            data: "get-root",
+        });
+    }),
+
+    http.post(`${PUBLIC_API_BASE_URL}/echo`, async ({ request }) => {
+        return HttpResponse.json(await request.json());
+    }),
+];
+
+const server = setupServer(...handlers);
+
+// Start server before all tests
+beforeAll(() => server.listen());
+
+// Close server after all tests
+afterAll(() => server.close());
+
+// Reset handlers after each test. This is important for test isolation
+afterEach(() => server.resetHandlers());
+
+describe("HttpClient", () => {
+    test("When get request is called, then correct data is returned", async () => {
+        const httpClient = new HttpClient("test");
+
+        const response = await httpClient.get();
+
+        expect(response).toBeDefined();
+        expect(response.status).toBe(200);
+        expect(await response.json()).toEqual({ data: "get-test" });
+    });
+
+    test("When post request is called, then correct data is returned", async () => {
+        const httpClient = new HttpClient("test");
+
+        const response = await httpClient.post();
+
+        expect(response).toBeDefined();
+        expect(response.status).toBe(200);
+        expect(await response.json()).toEqual({ data: "post-test" });
+    });
+
+    test("When put request is called, then correct data is returned", async () => {
+        const httpClient = new HttpClient("test");
+
+        const response = await httpClient.put();
+
+        expect(response).toBeDefined();
+        expect(response.status).toBe(200);
+        expect(await response.json()).toEqual({ data: "put-test" });
+    });
+
+    test("When patch request is called, then correct data is returned", async () => {
+        const httpClient = new HttpClient("test");
+
+        const response = await httpClient.patch();
+
+        expect(response).toBeDefined();
+        expect(response.status).toBe(200);
+        expect(await response.json()).toEqual({ data: "patch-test" });
+    });
+
+    test("When delete request is called, then correct data is returned", async () => {
+        const httpClient = new HttpClient("test");
+
+        const response = await httpClient.delete();
+
+        expect(response).toBeDefined();
+        expect(response.status).toBe(200);
+        expect(await response.json()).toEqual({ data: "delete-test" });
+    });
+
+    test("When get request is called without path, then correct data is returned", async () => {
+        const httpClient = new HttpClient();
+
+        const response = await httpClient.get();
+
+        expect(response).toBeDefined();
+        expect(response.status).toBe(200);
+        expect(await response.json()).toEqual({ data: "get-root" });
+    });
+
+    test("When get request is called with body, then correct data is returned", async () => {
```

Comment:
```suggestion
    test("When post request is called with body, then correct data is returned", async () => {
```

Your GET requests do not have a body ;)